kolibri/plugins/setup_wizard/assets/src/views/SetupWizardIndex.vue

```diff
@@ -50,7 +50,7 @@
     // and loses saved state
     beforeMount() {
       if (!this.$store.state.started) {
-        this.$router.replace('/');
+        this.$router.replace('/').catch(() => {});
```

Comment:
I saw this [comment](7068)

> This is caused by the router.js wrapper assigning the component name as a the route name. E.g. The parent route is given a name of UngroupedSetupSteps, even though it's not set in the routes array.
> 
> In the SetupWizard, we don't don't make the mistake of using to: {name: 'UngroupedSetupSteps'}, so we can ignore the warnings for now.

It's true that
```
Named Route 'UngroupedSetupSteps' has a default child route..
```
warning is caused by using a component name as a route name in our [router wrapper](https://github.com/learningequality/kolibri/blob/develop/kolibri/core/assets/src/router.js#L37). The `catch` introduced here doesn't remove the warning though (I suppose that it's reported even earlier) but I think it makes sense to leave the warning displayed so we are careful about potential problematic use-cases of the router wrapper in this regard.

The subsequent error
```
Uncaught (in promise) Error: Avoided redundant navigation to current location: "/".
```
should be actually displayed when you arrive on `setup_wizard/#/` because redundant navigation is happening (you visit `setup_wizard/#/` and then, because of the logic in `beforeMount` hook, you are redirected again to `setup_wizard/#/`). I would suggest avoiding swallowing errors in this way unless there is no other option so we won't run into unexpected issues not being reported. In this case it can be also solved by
```
if (!this.$store.state.started && this.$route.path !== '/') {
  this.$router.replace('/');
}
```

Reply:
Cool, that extra guard removes the error!